client/packages/system/src/Item/DetailView/Tabs/ItemVariants/BundledItemModal.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+
+import {
+  DialogButton,
+  InputWithLabelRow,
+  NumericTextInput,
+  Box,
+  useTranslation,
+  useDialog,
+  useKeyboardHeightAdjustment,
+  QueryParamsProvider,
+  createQueryParamsStore,
+  useNotification,
+  Typography,
+} from '@openmsupply-client/common';
+import {
+  BundledItemFragment,
+  DraftBundle,
+  ItemVariantFragment,
+  useUpsertBundledItem,
+} from '../../../api';
+import {
+  ItemVariantSearchInput,
+  StockItemSearchInput,
+} from '@openmsupply-client/system';
+
+export const BundledItemModal = ({
+  bundle,
+  variant,
+  onClose,
+}: {
+  bundle: BundledItemFragment | null;
+  variant: ItemVariantFragment;
+  onClose: () => void;
+}) => {
+  const t = useTranslation();
+  const { Modal } = useDialog({ isOpen: true, onClose, disableBackdrop: true });
+  const height = useKeyboardHeightAdjustment(350);
+  const { success } = useNotification();
+
+  const { draft, isComplete, updateDraft, save } = useUpsertBundledItem({
+    bundle,
+    principalVariant: variant,
+  });
+
+  return (
+    <Modal
+      title={t('title.bundled-with')}
+      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
+      okButton={
+        <DialogButton
+          disabled={!isComplete}
+          variant="ok"
+          onClick={() => {
+            save();
+            success(t('messages.item-variant-saved'))();
+            onClose();
+          }}
+        />
+      }
+      height={height}
+      width={700}
+      slideAnimation={false}
+    >
+      <QueryParamsProvider
+        createStore={createQueryParamsStore({ initialSortBy: { key: 'name' } })}
+      >
+        <BundledItemForm
+          draft={draft}
+          updateDraft={updateDraft}
+          principalVariant={variant}
+        />
+      </QueryParamsProvider>
+    </Modal>
+  );
+};
+
+const BundledItemForm = ({
+  draft,
+  principalVariant,
+  updateDraft,
+}: {
+  draft: DraftBundle;
+  principalVariant: ItemVariantFragment;
+  updateDraft: (update: Partial<DraftBundle>) => void;
+}) => {
+  const t = useTranslation();
+
+  return (
+    <Box justifyContent="center" display="flex" gap={3}>
+      <Box display="flex" flexDirection="column" gap={1} flex={1}>
+        <InputWithLabelRow
+          label={t('label.item_one')}
+          labelWidth="200"
+          Input={
+            <Box width="100%">
+              <StockItemSearchInput
+                autoFocus={!draft.itemId}
+                openOnFocus={!draft.itemId}
+                onChange={item => updateDraft({ itemId: item?.id })}
+                currentItemId={draft.itemId}
+                extraFilter={item => item.id !== principalVariant.itemId}
+              />
```

Comment:
How hard would it be to filter the item list to only show ones that have Variants defined?

I guess you'd need to update the possible filters for Items, so might be a bit much for now. I think it would be useful though -- it seems a little annoying to be able to select an item and then discover it has not variants that can be bundled. Maybe a new issue?

Reply:
Hm yeah nice idea, but yep would be quite a bit of extra work... I also don't know whether it would be confusing to only see a subset of the items? Would need to explain somewhere that only items with variants are available in the list 🤷‍♀️ 

Thinking it's okay for now, given this is only on Central Server, and only used for GAPS, where we're expecting a known data set. Future idea is to actually have "default" item variant, for any items that don't have variants defined, which would solve this issue 🙏